drivers/flash/soc_flash_nrf.h

```diff
@@ -0,0 +1,173 @@
+/*
+ * Copyright (c) 2020 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#ifndef __SOC_FLASH_NRF_H__
+#define __SOC_FLASH_NRF_H__
+
+#include <kernel.h>
+
+#define FLASH_OP_DONE    (0) /* 0 for compliance with the driver API. */
+#define FLASH_OP_ONGOING (-1)
+
+struct flash_context {
+	uint32_t data_addr;  /* Address of data to write. */
+	uint32_t flash_addr; /* Address of flash to write or erase. */
+	uint32_t len;        /* Size off data to write or erase [B]. */
+#ifndef CONFIG_SOC_FLASH_NRF_RADIO_SYNC_NONE
+	uint8_t  enable_time_limit; /* set execution limited to the execution
+				     * window.
+				     */
+#endif /* !CONFIG_SOC_FLASH_NRF_RADIO_SYNC_NONE */
+#if defined(CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE)
+	uint32_t flash_addr_next;
+#endif /* CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE */
+}; /*< Context type for f. @ref write_op @ref erase_op */
+
+#ifndef CONFIG_SOC_FLASH_NRF_RADIO_SYNC_NONE
+
+
+#if defined(CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE)
+/* The timeout is multiplied by 1.5 because switching tasks may take
+ * significant portion of time.
+ */
+#define FLASH_TIMEOUT_MS ((FLASH_PAGE_ERASE_MAX_TIME_US) * \
+			  (FLASH_PAGE_MAX_CNT) / 1000 * 15 / 10)
+#else
+
+#define FLASH_TIMEOUT_MS ((FLASH_PAGE_ERASE_MAX_TIME_US) * \
+			  (FLASH_PAGE_MAX_CNT) / 1000)
+#endif /* CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE */
+
+
+#if CONFIG_SOC_FLASH_NRF_RADIO_SYNC
+
+#define FLASH_RADIO_ABORT_DELAY_US 1500
+#define FLASH_RADIO_WORK_DELAY_US  200
+
+/* delay needed for start execution-window */
+#define FLASH_SYNCH_SWITCHING_TIME (FLASH_RADIO_ABORT_DELAY_US +\
+				   FLASH_RADIO_WORK_DELAY_US)
+#endif
+
+/**
+ * @defgroup nrf_flash_synch synch backend API
+ *
+ * API declared bellow contains prototypes of function which shall be
+ * implemented by the synchronization backend.
+ * @{
+ */
+
+/**
+ * Calback which executes the flash operation.
+ *
+ * @param context pointer to flash_context structure.
+ * @retval @ref FLASH_OP_DONE once operation was done, @ref FLASH_OP_ONGOING if
+ *         opperation needs more time for execution.
+ */
+typedef int (*flash_op_handler_t) (void *context);
+
+struct flash_op_desc {
+	flash_op_handler_t handler;
+	struct flash_context *context; /* [in,out] */
+};
+
+/**
+ * Synchronization backend driver initialization procedure.
+ *
+ * This will be run within flash driver initialization
+ */
+int nrfsb_synch_backend_init(void);
+
+/**
+ * Set synchronization context for synchronous operations.
+ *
+ * This function set beckend's internal context for expected timing parameters.
+ *
+ * @param duration Duration of the execution window [us]
+ * @param interval Interval beatwen execution windows [us]
```

Comment:
Is `interval` required parameter? Is it required that the backend executes flash operations in given interval? What is accepted accuracy of this interval?

I thought that the backend shall execute flash operations as frequently as possible, as soon as possible. Am I mistaken?

Reply:
Ticker backed need to know interval (because of ticker/ll design).
But right, in general this could be implementation part to provide this interval.
Will remove interval parameter.

Reply:
Done.
@de-nordic  Actually this change timing while partial erase operation:
previously interval was ~90 ms, and slot was partial erase CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE_MS ms
now interval is the same as slot, so CONFIG_SOC_FLASH_NRF_PARTIAL_ERASE_MS ms

I believe such a long and disproportionate interval was kept by oversight.
I retested the driver for very short partial erase time (3 ms).